libsolidity/analysis/TypeChecker.cpp

```diff
@@ -2781,12 +2781,53 @@ bool TypeChecker::visit(Identifier const& _identifier)
 	IdentifierAnnotation& annotation = _identifier.annotation();
 	if (!annotation.referencedDeclaration)
 	{
+		vector<Declaration const*>& uniqueDeclarations = annotation.overloadedDeclarations;
```

Comment:
Does this code run later now that it's in the `TypeChecker`?

Reply:
It indeed does - now it's run only right before it's needed.
I don't really think this has drawbacks, though - the only thing this might affect is the AST in error cases not containing the ``overloadedDeclarations`` information...

Reply:
Ok!